python/pyarrow/src/arrow-python.pc.in

```diff
@@ -1,28 +0,0 @@
-# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Because we can't provide `.pc` in portable way for Python package.
`.pc` requires fixed install location but Python package is relocatable. Users can install it to `/usr/local/lib/python*/dist-packages/pyarrow/`, `~/.local/lib/python*/site-packages/pyarrow/` and so on.
If we want to keep `.pc` support, we need to rewrite `.pc` after pyarrow is installed. (This is the approach that is used by MSYS2 package.)

FYI: `.pc` aren't installed since our `cpp/src/arrow/python/` -> `python/` migration. It means that users already can't use `arrow-python.pc` since 10.0.0.

Reply:
the conda builds don't currently [test](https://github.com/apache/arrow/blob/master/dev/tasks/conda-recipes/arrow-cpp/meta.yaml#L132-L167) for the pc files, but they _are_ being [installed](https://github.com/regro/libcfgraph/blob/master/artifacts/libarrow/conda-forge/linux-64/libarrow-10.0.1-hd014966_3_cpu.json#L809-L821) even for v>=10.

I think they should continue to be installed at least on conda, where the location is well-specified (and conda will take care of fixing the `.pc` files automatically).

Reply:
@h-vetinari what you point to are the .pc files for libarrow, and that doesn't include the arrow_python library (anymore, since 10.0). This PR only affects the latter, and not the pc files for libarrow.

Reply:
I have a test lined up for pkg-config metadata in conda builds that I can either push into https://github.com/apache/arrow/pull/15014, but more likely would put in a separate PR after that one is merged (or into this one, if desired).

In any case, I think this PR should ideally be tested to not break the conda build setup (I'm happy to help with eventual adaptations).

Reply:
[gh hadn't updated the page so I didn't see you comment while writing my follow-up, apologies]

Sorry I overlooked what `.pc` files this was pointing at. It seems there are still some `.pc` files being [installed](https://github.com/regro/libcfgraph/blob/master/artifacts/pyarrow/conda-forge/linux-64/pyarrow-10.0.1-py38h03a3090_3_cpu.json#L416-L417) for `pyarrow` (or rather `.pc.in`, so probably not functional).

I don't have very strong feelings about `.pc` files for the python libs, but in general I think if we can create them easily (and it's not an undue maintenance burden), we should keep them for the conda builds.

Reply:
And to complement to what @kou said: this is very similar for numpy. There one also need to use `np.get_include()`.